shared-bindings/keypad/ShiftRegisterKeys.c

```diff
@@ -155,7 +200,7 @@ STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(keypad_shiftregisterkeys___exit___obj
 //|         ...
 
 //|     key_count: int
-//|     """The number of keys that are being scanned. (read-only)
+//|     """The total number of keys that are being scanned. (read-only)
```

Comment:
For the user code, would it be more useful to return a tuple in the case of multiple shift registers? Instead of doing a gc_alloc, you could create a tuple, and you can index into that, and return it for `.key_count`. (I was thinking about a new `.key_counts`, but the input arg is still `key_count`, so it would not be consistent.)

Reply:
How important is it to have a different key count per shift register? Is this needed for a real in-the-wild design, or is it more a hypothetical?

Another alternative would be to have a single argument giving the maximum count of keys. "shift register 0" would be 0..(n-1), "shift register 1" would be n..(2*n-1) and so on. Unused positions _would_ need to be hard-wired to the non-pressed state (not an undefined state or the pressed state), and some key numbers would be unused if the real number of keys per shift register is "ragged" (e.g., 3, 5, 3, 6, or whatever).

Reply:
The current use case i have is ragged, but I could live in my case with a single key_count and ignore unused keys (which would i my case be signalled as pressed and end up in the event queue). I believe the key_count() function is also used to allocate the currently_pressed / previously_pressed stuff for the toplevel keypad code, so a tuple might not work.

Reply:
If this is needed to support already existing hardware then I'm not opposed to incorporating a feature that supports ragged shift registers.